ui/src/app/workflows/components/workflow-logs-viewer/workflow-logs-viewer.tsx

```diff
@@ -98,15 +99,14 @@ export class WorkflowLogsViewer extends React.Component<WorkflowLogsViewerProps,
 
     private refreshStream(): void {
         this.ensureUnsubscribed();
+
+        this.setState({lines: [], loaded: false, error: undefined});
+
         this.subscription = services.workflows.getContainerLogs(this.props.workflow, this.props.nodeId, this.props.container, this.props.archived).subscribe(
             log => {
                 this.setState(state => {
                     const newState = {...state, loaded: true};
-
-                    if (log) {
-                        newState.lines = newState.lines.concat(log.split('\n'));
-                    }
-
+                    newState.lines.push(log + '\n');
```

Comment:
With the refactor to the observables, each value that is emitted is guaranteed to be a single line of logs (rather than potentially being undefined), so we no longer need to guard against that case.